### ISSUE TITLE
 py setup.py install failing 

### ISSUE THREAD

**Issue by jayadevmit** [jpype-project/jpype#1028]
while trying to install Jpype , the setup fails .Tried different option like using pip  / setup.py after downloading the artifacts from nexus  etc

 py setup.py install

Falling back to provided JNI headers, since your provided JAVA_HOME "C:\Program Files\Java\jdk-17.0.1" does not provide jni.h
Falling back to provided JNI headers, since your provided JAVA_HOME "C:\Program Files\Java\jdk-17.0.1" does not provide jni.h
Falling back to provided JNI headers, since your provided JAVA_HOME "C:\Program Files\Java\jdk-17.0.1" does not provide jni.h
running install
C:\Users\Jay\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
C:\Users\Jay\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\command\easy_install.py:156: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running bdist_egg
running egg_info
package init file 'jpype\_pyinstaller\__init__.py' not found (or not a regular file)
warning: no files found matching '*.class' under directory 'native'
writing manifest file 'JPype1.egg-info\SOURCES.txt'
running install_lib
running build_py
running build_ext
Call build extensions
Call build ext
"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.30.30705\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DWIN32=1 -Inative\common\include -Inative\python\include -Inative\embedded\include "-IC:\Program Files\Java\jdk-17.0.1\include" "-IC:\Program Files\Java\jdk-17.0.1\include\win32" -IC:\Users\Jay\AppData\Local\Programs\Python\Python310\include -IC:\Users\Jay\AppData\Local\Programs\Python\Python310\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.30.30705\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /EHsc /Tpnative\common\jp_array.cpp /Fobuild\temp.win-amd64-3.10\Release\native\common\jp_array.obj /Zi /EHsc /std:c++14
jp_array.cpp
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\python\include\jp_pythontypes.h(284): error C2061: syntax error: identifier 'ssize_t'
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\python\include\jp_pythontypes.h(285): error C2805: binary 'operator [' has too few parameters
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\python\include\jp_pythontypes.h(285): error C2333: 'JPPyObjectVector::operator []': error in function declaration; skipping function body
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\common\include\jp_primitivetype.h(44): error C3646: 'getItemSize': unknown override specifier
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\common\include\jp_primitivetype.h(44): error C2059: syntax error: ')'
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\common\include\jp_primitivetype.h(44): error C2238: unexpected token(s) preceding ';'
native\common\jp_array.cpp(181): error C2065: 'ssize_t': undeclared identifier
native\common\jp_array.cpp(181): error C2146: syntax error: missing ';' before identifier 'itemsize'
native\common\jp_array.cpp(181): error C2065: 'itemsize': undeclared identifier
native\common\jp_array.cpp(182): error C2065: 'ssize_t': undeclared identifier
native\common\jp_array.cpp(182): error C2146: syntax error: missing ';' before identifier 'sz'
native\common\jp_array.cpp(182): error C2065: 'sz': undeclared identifier
native\common\jp_array.cpp(188): error C2065: 'itemsize': undeclared identifier
native\common\jp_array.cpp(188): error C2039: 'getItemSize': is not a member of 'JPPrimitiveType'
C:\Users\Jay\Desktop\WorkSpace\php_jira_worklog\jpel\native\common\include\jp_primitivetype.h(20): note: see declaration of 'JPPrimitiveType'
native\common\jp_array.cpp(189): error C2065: 'sz': undeclared identifier
native\common\jp_array.cpp(189): error C2065: 'itemsize': undeclared identifier
native\common\jp_array.cpp(193): error C2065: 'sz': undeclared identifier
native\common\jp_array.cpp(197): error C2065: 'itemsize': undeclared identifier
native\common\jp_array.cpp(206): error C2065: 'sz': undeclared identifier
native\common\jp_array.cpp(216): error C2065: 'itemsize': undeclared identifier
native\common\jp_array.cpp(224): error C2065: 'itemsize': undeclared identifier
native\common\jp_array.cpp(227): error C2065: 'sz': undeclared identifier
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.30.30705\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

**Comment by john-hen** [comment]
Not sure what those "artifacts from nexus" are, but this is an issue with Python 3.10 that has been fixed, see #1009. There has not been a release yet since then, but what's currently in `master` will build on Windows.

**Comment by Thrameos** [comment]
Yeah, we should release 1.4 with that fix soon.

**Comment by reinvantveer** [comment]
Thanks @Thrameos that would be very helpful!

**Comment by reinvantveer** [comment]
For the impatient: you may be able to work around this using
```
pip install https://github.com/jpype-project/jpype.git@cfd5f84f14ddc6d551460b5ce96b22e4ff754d5c
```
where cfd5f84f14ddc6d551460b5ce96b22e4ff754d5c is the current (at the moment of writing this up) commit hash of the master branch. The necessary changes are already in the master branch, but this works only if JPype is not a version-pinned subdependency of one of your project dependencies.

But note that pip-installing from a git repo is always more of a maintainance burden than from the canonical [cheese shop](https://pypi.org)

**Comment by JacksonMaxfield** [comment]
Hey @Thrameos I saw the new 1.4.0 release on GitHub! I can now install via tag :tada: :heart:  Thanks for all the work put in on this!

Any timeline on pypi release?

**Comment by Thrameos** [comment]
Shortly..  see #1063

**Comment by JacksonMaxfield** [comment]
Thank you @Thrameos! I think this can be closed? My tests now pass with 1.4.0 from pypi! :tada:
